### ISSUE TITLE
Simplify configuration

### ISSUE THREAD

**Issue by justinjstark** [justinjstark/SmtpRouter#9]
I'm trying to hit the balance of using Dependency Injection and allowing for explicit stack configuration. Each middleware should be able to use dependencies that are injected (ex: an HttpPost middleware that uses IHttpClientFactory). This requires sending the IServiceProvider down to the stack configuration (which makes it U-G-L-Y) or using a completely different technique to set up a Stack (I'm looking for patterns).

The bottom line is there are certain dependencies which we expect to be injected (ILogger, IHttpClientFactory, etc) and some parameters we want to inject during Stack configuration (Log formatter, rerouteRules, etc).

**Comment by justinjstark** [comment]
The bottom line is there are certain dependencies which we expect to be injected (ILogger<T>, IHttpClientFactory, etc) and some parameters we want to inject during Stack configuration (Log formatter, rerouteRules, etc)
